pkg/trace/info/stats.go

```diff
@@ -155,15 +189,66 @@ func (ts *TagStats) publish() {
 	metrics.Count("datadog.trace_agent.receiver.events_extracted", eventsExtracted, tags, 1)
 	metrics.Count("datadog.trace_agent.receiver.events_sampled", eventsSampled, tags, 1)
 	metrics.Count("datadog.trace_agent.receiver.payload_accepted", requestsMade, tags, 1)
+
+
+	var droppedReasons []string
+	for _, c := range ts.droppedTraceReasonCounts() {
+		if c.count > 0 {
+			metrics.Count("datadog.trace_agent.normalizer.traces_dropped", c.count, append(tags, "reason:" + c.reason), 1)
+			droppedReasons = append(droppedReasons, c.reason + ":" + strconv.FormatInt(c.count, 10))
+		}
+	}
+
+	var malformedReasons []string
+	for _, c := range ts.malformedTraceReasonCounts() {
+		if c.count > 0 {
+			metrics.Count("datadog.trace_agent.normalizer.traces_malformed", c.count, append(tags, "reason:" + c.reason), 1)
+			malformedReasons = append(malformedReasons, c.reason + ":" + strconv.FormatInt(c.count, 10))
+		}
+	}
+
+	var normalizerMessages []string
```

Comment:
Can you please remove this entire logging logic. This is not the right place to do it. Let's let `publish` publish only stats. The receiver stats are published once per minute [here](https://github.com/DataDog/datadog-agent/blob/bd9c45ba2f055b9349cc968884c498776279d24e/pkg/trace/api/api.go#L395-L397) and they are the result of the `fmt.Stringer` implementation from [here](https://github.com/DataDog/datadog-agent/blob/bd9c45ba2f055b9349cc968884c498776279d24e/pkg/trace/info/stats.go#L246). All you have to do is improve the `String` method a bit to include dropped and malformed traces and it'll be 👍

Reply:
Yeah I planned on moving it elsewhere anyway, just had it there while I was working on it